app/workers/series_worker.rb

```diff
@@ -133,10 +137,11 @@ def perform(series_id, series_size)
         redis.decr keys[:busy_workers]
       end
     end
+    break ## end of loop do - only run once
+    end
+    logger.info "SIDEKIQ perform: finished with series #{series_id}"
+    failure = SidekiqFailure.find_by(series_id: series_id)
+    failure.destroy if failure
```

Comment:
What happens if a series has two source. the first fails creating a SidekiqFailure. The second source succeeds causing the SidekiqFailure to be destroyed. Does it make a sound?

Reply:
Good point.